### PR TITLE
Add Gulp/BrowserSync dev workflow

### PR DESCRIPTION
**Pull request by rphi**
BrowserSync is nice for auto-reload on changes and mobile device testing. Plus Gulp opens up the possibility of minifying js and auto prefixing css/sass.

### CODE REVIEW

File: _config.yml [hunk 1]
```diff
@@ -18,25 +20,25 @@ collections:
...
-    values:
-      layout: "default"
+  # - scope:
+  #     path: "" # an empty string here means all files in the project
+  #   values:
+  #     layout: "default"
 
   - scope:
-      path: ""
+      path: "_minutes"
```
Comment (qaisjp) [inline _config.yml]: Are you sure the `path` field is required at all?  It piggybacks off the collections feature.

File: _config.yml [hunk 2]
```diff
@@ -4,7 +4,9 @@ sass:
    sourcemap: true
 
 url: "http://comp-soc.com"
-baseurl: "/compsoc-website" # does not include hostname
+baseurl: "" # does not include hostname
```
Comment (qaisjp) [inline _config.yml]: `baseurl` regression

File: _config.yml [hunk 3]
```diff
@@ -18,25 +20,25 @@ collections:
...
     
 defaults:
-  - scope:
-      path: "" # an empty string here means all files in the project
-    values:
-      layout: "default"
+  # - scope:
+  #     path: "" # an empty string here means all files in the project
+  #   values:
+  #     layout: "default"
```
Comment (qaisjp) [inline _config.yml]: What is the purpose of commenting this? Consider removing completely, instead of commenting out.

File: _config.yml (hunk 2 above)
Comment (rphi) [inline _config.yml]: Nope, this was so that the browser sync webserver would work

File: _config.yml (hunk 3 above)
Comment (rphi) [inline _config.yml]: Yup, I was trying to stop Jekyll from processing the gulp stuff, probably could be deleted

File: _config.yml (hunk 1 above)
Comment (rphi) [inline _config.yml]: Not entirely sure, but it reduces the chance of anything getting miscategorised?

**Comment by rphi** [comment]
Hah, and that email doesn't go anywhere because I broke my mx records.... at least it isn't rob@rob-mb.gg.rphi.uk

**Comment by qaisjp** [comment]
yep, and your git email is actually set to `rob[@]phii.uk`, including those `[` and `]` :P

Also, the "breaking up the PRs" bit was related to the config yml file (which i've included [here](https://github.com/qaisjp/compsoc-website/commit/0c67f6dc9e36398b012640f2e6f0eb026494935d)).

Webpack is a rabbit hole that is a complete pain to set up, especially if you haven't used it before. I would recommend you not to bother with it unless you're sure you want to use it later (and it's worth practicing with it on a demo project before applying it here).

`yarn` is almost a replacement for `npm`, yes.

**Pull request closed without merging**